src/node_i18n.cc

```diff
@@ -501,19 +500,30 @@ void ConverterObject::Decode(const FunctionCallbackInfo<Value>& args) {
         converter->set_bom_seen(true);
       }
     }
-    ret = ToBufferEndian(env, &result);
-    if (omit_initial_bom && !ret.IsEmpty()) {
+
+    Local<Value> error;
+    char16_t* output = result.out();
+    size_t beginning = 0;
+    size_t length = 0;
+
+    for (size_t i = 0; i < result.length(); i++) length += sizeof(output[i]);
```

Comment:
Suggestion: Simplify this to

```suggestion
    size_t length = result.length() * sizeof(UChar);
```

practically speaking, compilers will just do this anyway, but I think it would be a bit more idiomatic to not spell this out as a for loop since `sizeof(output[i])` is a constant.

Reply:
Nice! I'll update the pull request